### PR TITLE
RESTClient: docs: add the troubleshooting section

### PR DESCRIPTION
**Pull request by burnash**

### CODE REVIEW

File: docs/website/docs/general-usage/http/rest-client.md [hunk 1]
```diff
@@ -527,4 +527,70 @@ from dlt.sources.helpers.rest_client import paginate
...
+```
+
+### `RESTClient.paginate()`
+
+Debugging `paginate()` is trickier because it's a generator function that yields [`PageData`](#pagedata) objects. Here's several ways to debug the `paginate()` method:
+
+1. Enable [logging](../../running-in-production/running.md#set-the-log-level-and-format) to see detailed information about the HTTP requests:
+
+```bash
+RUNTIME__LOG_LEVEL=INFO python my_script.py
```
Comment (sultaniman) [inline docs/website/docs/general-usage/http/rest-client.md:559]: is `INFO` sufficient here?

**Review by sultaniman — APPROVED** [review]
LGTM 🚀 imo we need to improve exceptions and logging as @rudolfix already did maybe we can do extended logging and rich formatted exception logs if `level=DEBUG`?

File: docs/website/docs/general-usage/http/rest-client.md (hunk 1 above)
Comment (burnash) [reply docs/website/docs/general-usage/http/rest-client.md:559]: Yes, this is enough to see the requests info

**Comment by burnash** [comment]
Good point @sultaniman. Do you have good examples of rich formatting the exceptions? I think one thing that would be great is to see the actual response content in case of HTTP exceptions, as sometimes there's required context that can clarify the error.

**Comment by sultaniman** [comment]
@burnash I only know about rich traceback printer https://rich.readthedocs.io/en/stable/traceback.html

**Pull request merged**